### PR TITLE
Added workflow for updating backstage

### PR DESCRIPTION
**Pull request by keyluck**
Added a workflow for updating backstage, creating a PR, and logging the output of the update to the PR

### CODE REVIEW

**Review by keyluck — COMMENTED** [review]
Reviewed changes for backstage update using github actions

File: .github/workflows/backstage-update.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    needs: check-for-existing-update
+    if: needs.check-for-existing-update.outputs.exists != '[ ]' 
+    steps:
+    - name: Failed to Update Backstage
+      uses: actions/github-script@v3
+      with:
+        script: |
+          console.log("Failed to Update Backstage - Pull Request already exists: ",${{ needs.check-for-existing-update.outputs.url }})
+      
+
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:39]:
```suggestion
```

File: .github/workflows/backstage-update.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        url=$(echo $exists | jq .[] | jq .html_url)
+        echo ::set-output name=exists::$exists
+        echo ::set-output name=url::$url
+        
+  link-pr:
+    runs-on: ubuntu-latest
+    needs: check-for-existing-update
+    if: needs.check-for-existing-update.outputs.exists != '[ ]' 
+    steps:
+    - name: Failed to Update Backstage
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:33]:
```suggestion
    - name: Failed to update Backstage
```

File: .github/workflows/backstage-update.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: |
+          yarn install
+      - name: Update Backstage
+        run: |
+          npm run backstage-update
+      - name: Compare Create App versions
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:53]:
```suggestion
      - name: Compare & create app versions
```

File: .github/workflows/backstage-update.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        run: |
+          body=$(sed -n '/breaking changes/, $p' backstage-update-log.txt)
+          body="${body//$'\n'/'%0A'}"
+          echo ::set-output name=body::$body
+      - name: Set output variables
+        id: vars
+        run: |
+          echo ::set-output name=pr_title::"Update Backstage (auto-generated)"
+          echo ::set-output name=date::$(date +'%Y-%m-%d')
+      - name: Create Pull Request
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:71]:
```suggestion
      - name: Create PR
```

File: .github/workflows/backstage-update.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        id: cpr
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:72]:
```suggestion
        id: create-pr
```
This doesn't have to be overly terse. Feel free to call it something else.

File: .github/workflows/backstage-update.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            labels: |
+              automated
+              Pull Request
+              backstage
+              update
+            reviewers: lighthouse-bandicoot
+            draft: false
+      - name: Check outputs
+        run: |
+          echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:102]:
```suggestion
          echo "Pull Request Number - ${{ steps.create-pr.outputs.pull-request-number }}"
```

File: .github/workflows/backstage-update.yml [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+          echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
```
Comment (rianfowler) [inline .github/workflows/backstage-update.yml:103]:
```suggestion
          echo "Pull Request URL - ${{ steps.create-pr.outputs.pull-request-url }}"
```

**Review by rianfowler — APPROVED** [review]
Looks great. Had some minor nits. Feel free to accept or ignore.

File: .github/workflows/backstage-update.yml [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+      run: |
+        exists=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/department-of-veterans-affairs/lighthouse-backstage/pulls?head=department-of-veterans-affairs:auto-update-backstage)
+        url=$(echo $exists | jq .[] | jq .html_url)
+        echo ::set-output name=exists::$exists
+        echo ::set-output name=url::$url
+        
+  link-pr:
+    runs-on: ubuntu-latest
+    needs: check-for-existing-update
+    if: needs.check-for-existing-update.outputs.exists != '[ ]'
```
Comment (KaemonIsland) [inline .github/workflows/backstage-update.yml:31]:
```suggestion
    if: needs.check-for-existing-update.outputs.exists != '[ ]'
```

Extra space at the end.

File: .github/workflows/backstage-update.yml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      uses: actions/github-script@v3
+      with:
+        script: |
+          console.log("Failed to Update Backstage - Pull Request already exists: ",${{ needs.check-for-existing-update.outputs.url }})
+      
+ 
+  update-backstage:
+    runs-on: ubuntu-latest
+    needs: check-for-existing-update
+    if: needs.check-for-existing-update.outputs.exists == '[ ]'
```
Comment (KaemonIsland) [inline .github/workflows/backstage-update.yml:42]:
```suggestion
    if: needs.check-for-existing-update.outputs.exists == '[ ]'
```

Extra space here as well.

File: .github/workflows/backstage-update.yml [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+ 
+  update-backstage:
+    runs-on: ubuntu-latest
+    needs: check-for-existing-update
+    if: needs.check-for-existing-update.outputs.exists == '[ ]' 
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: |
```
Comment (KaemonIsland) [inline .github/workflows/backstage-update.yml:47]: Is there a reason we want to include the `|` for single-line commands?

File: docs/backstage-update.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+# Automated Backstage Update
```
Comment (KaemonIsland) [inline docs/backstage-update.md:1]:
```suggestion
# Automated Backstage Update
```

File: docs/backstage-update.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
+# Automated Backstage Update 
+
+[Backstage Update Workflow file](../.github/workflows/backstage-update.yml)
+
+# Update Backstage Workflow Jobs
+
+## check-for-existing-update
+Description: Before upgrade process begins, this job checks if there is currently an open pull request to update backstage. 
+    
+Steps:
```
Comment (KaemonIsland) [inline docs/backstage-update.md:10]: Could you wrap this section in `Code formatting?` It'll make it a lot easier to read. 👍 

You can view the generated docs [here.](https://department-of-veterans-affairs.github.io/lighthouse-backstage/backstage-update/)

File: docs/backstage-update.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    - Uses: [actions/checkout@v2](https://github.com/actions/checkout)
+    - Checkout github repository so workflow can access it
+  - Check for existing auto-update PR
+    - Uses command line to make cURL request to GitHub API for all open pull requests using the "auto-update-backstage" branch
+    - Stores url to open pull request as output
+
+## link-pr
+Description: If there is an open pull request to update backstage, this job will output a link to the open pull request.
+
+Steps:
```
Comment (KaemonIsland) [inline docs/backstage-update.md:21]: Same here with `code formatting.`

File: docs/backstage-update.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  - Failed to Update Backstage
+    - Uses: [actions/github-script@v3](https://github.com/actions/github-script)
+    - Outputs the open pull request url retrieved from GitHub API request
+
+## update-backstage
+Description: If there is no open pull request to update backstage, this job will create a branch, perform the upgrade process, and create a pull request with the new changes.
+
+Steps:
```
Comment (KaemonIsland) [inline docs/backstage-update.md:29]: And here as well.

**Review by KaemonIsland — APPROVED** [review]

**Comment by KaemonIsland** [comment]
Most of my review comments are nitpicking about spacing/formatting. The code looks great!

One more note I had, would you mind adding your know `docs` file to [mkdocks.yml](https://github.com/department-of-veterans-affairs/lighthouse-backstage/blob/main/mkdocs.yml#L4) That way we can get a link to view your documentation. 👍

File: .github/workflows/backstage-update.yml (hunk 10 above)
Comment (keyluck) [reply .github/workflows/backstage-update.yml]: Removed | from single line commands

**Pull request merged**